airflow/providers/amazon/aws/hooks/redshift.py

```diff
@@ -138,6 +138,26 @@ def create_cluster_snapshot(self, snapshot_identifier: str, cluster_identifier:
         )
         return response['Snapshot'] if response['Snapshot'] else None
 
+    def wait_for_state(self, cluster_identifier: str, target_state: str, check_interval: float) -> None:
```

Comment:
Thanks for the contribution!

Instead of writing this wait method in the hook and injecting it inside many of the operators you can instead create a sensor for waiting for states. That sensor can then be added to dags as needed. It's a bit more graceful of a solution especially when combined with something like Smart Sensors.

You can see an example in the recently developed EKS code. You can see the sensor here:
https://github.com/apache/airflow/blob/752575cb84a28c41723b5151fd6451a72f4da5fb/airflow/providers/amazon/aws/sensors/eks.py#L54

And you can see it's usage in the EKS example dags, for example:
https://github.com/apache/airflow/blob/752575cb84a28c41723b5151fd6451a72f4da5fb/airflow/providers/amazon/aws/example_dags/example_eks_with_nodegroup_in_one_step.py#L93

Reply:
Thanks @o-nikolas! I was trying to follow the same pattern as the EC2 modules:
`airflow/providers/amazon/aws/hooks/ec2.py`
and
`airflow/providers/amazon/aws/operators/ec2_start_instance.py`

would you still prefer if i follow the eks example? 

I kind of feel like I could develop the Sensor, but maybe leave the `wait_for_state` in the Operator also? as a safe check? what do you think?

Reply:
There is a design pattern in several operators that include the option, rather than a forced action, to wait for termination/completion if desired. 
https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/operators/data_factory.py
https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/dataflow.py
https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/dataproc.py
https://github.com/apache/airflow/blob/main/airflow/providers/tableau/operators/tableau_refresh_workbook.py

This way users could choose whether to wait for the task to complete as part of the operator or use a sensor to poke for status (mainly useful for long-running actions). Of course, Deferrable Operators solve for this too 🙂

Reply:
got it thanks @josh-fell, the latest commit fixes this and removes the `wait_for_state` and this way users can choose to use the sensor (already available).